drivers/interrupt_controller: stm32: Use LL API

erwango opened the pull request:
exti driver implementation does not fit all SoCs because
some EXTI ip does not match stm32_exti register map provided.
Instead of providing exti register map for all SoCs, use LL API
which abstracts IP variations and enable uniform use of the drivers
on all STM32SoCs.

Tested on F1/F3/F4/F7/L0/L4

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

ydamigos commented on drivers/interrupt_controller/exti_stm32.c:
```diff
@@ -194,9 +173,16 @@ void stm32_exti_disable(int line)
+		return LL_EXTI_IsActiveFlag_0_31(1 << line);
+	} else {
+#if EXTI_LINES > 32
+		return LL_EXTI_IsActiveFlag_0_31(1 << (line - 32));
```
LL_EXTI_IsActiveFlag_32_63?

erwango replied:
Exact! updated, txs!

ydamigos reviewed (changes requested):
Looks good. A minor tweak.

erwango committed:
drivers/interrupt_controller: stm32: Use LL API

exti driver implementation does not fit all SoCs because
some EXTI ip does not match stm32_exti register map provided.
Instead of providing exti register map for all SoCs, use LL API
which abstracts IP variations and enable uniform use of the drivers
on all STM32SoCs.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>
